c/src/ml-api-service-agent-client.c

```diff
@@ -143,9 +143,8 @@ ml_service_get_pipeline (const char *name, char **pipeline_desc)
         "The parameter, 'name' is NULL, It should be a valid string");
   }
 
-  if (!pipeline_desc) {
-    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-        "The parameter 'pipeline_desc'. It should be a valid char**");
+  if (pipeline_desc) {
+    *pipeline_desc = NULL;
```

Comment:
I don't know well.
Shouldn't 'pipeline_desc' be the first address where memory is allocated?

Reply:
This is a **[out]** parameter.

An example usage as follows:
```
char *desc;

ml_service_get_pipeline(name, &desc);
```

```desc``` will point the first block of an allocated memory inside wherever ```ml_service_get_pipeline```.

Reply:
Thank you for the detailed description.